pkgs/top-level/all-packages.nix

```diff
@@ -15339,6 +15339,8 @@ with pkgs;
   julia-stable-bin = julia_18-bin;
   julia-bin = julia-stable-bin;
 
+  juliaWithPackages = callPackage ../development/julia-modules { };
```

Comment:
is there a particular reason (I did not see one while going through the code) to implement it as `juliaWithPackages` not as an attribute for the (supported julia versions) like for `python.withPackages`?

Reply:
Nope! I would like to change it to be an attribute. When I looked at the main `julia` derivations it was a bit involved, but let me see if I can make that happen...

Reply:
Did you take a look at this PR: "https://github.com/NixOS/nixpkgs/pull/210685/files#diff-49d36bb134b21da2245f39b248e7fa295b4641094bc7e1fb41a565f738f0d957", maybe it helps.
I think you don't need to change the main derivations but create a function that adds the withPackages attribute (similar to what they do with wrap but without creating an actual wrapper).

Reply:
Okay, this is done in the latest commit